climada/util/api_client.py

```diff
@@ -486,3 +533,166 @@ def purge_cache(self, local_path):
         """
         dlf = Download.get(Download.path==str(local_path.absolute()))
         dlf.delete_instance()
+
+    def get_hazard(self, hazard_type, dump_dir=SYSTEM_DIR, **kwargs):
+        """Queries the data api for hazard datasets of the given type, downloads associated
+        hdf5 files and turns them into a climada.hazard.Hazard object.
+
+        Parameters
+        ----------
+        hazard_type : str
+            Type of climada hazard.
+        dump_dir : str, optional
+            Directory where the files should be downoladed. Default: SYSTEM_DIR
+            If the directory is the SYSTEM_DIR, the eventual target directory is organized into
+            dump_dir > hazard_type > dataset name > version
+        **kwargs :
+            additional parameters passed on to get_datasets
+
+        Returns
+        -------
+        climada.hazard.Hazard
+            The combined hazard object
+        """
+        if 'data_type' in kwargs:
+            raise ValueError("data_type is already given as hazard_type")
+        if not hazard_type in HAZ_TYPES:
+            raise ValueError("Valid hazard types are a subset of CLIMADA hazard types."
+                             f" Currently these types are supported: {HAZ_TYPES}")
+        datasets = self.get_datasets(data_type=hazard_type, **kwargs)
+        if not datasets:
+            raise ValueError("no datasets found meeting the requirements")
+        return self.to_hazard(datasets, dump_dir)
+
+    def to_hazard(self, datasets, dump_dir=SYSTEM_DIR):
+        """Downloads hdf5 files belonging to the given datasets reads them into Hazards and
+        concatenates them into a single climada.Hazard object.
+
+        Parameters
+        ----------
+        datasets : list of DatasetInfo
+            Datasets to download and read into climada.Hazard objects.
+        dump_dir : str, optional
+            Directory where the files should be downoladed. Default: SYSTEM_DIR
+            If the directory is the SYSTEM_DIR, the eventual target directory is organized into
+            dump_dir > hazard_type > dataset name > version
+
+        Returns
+        -------
+        climada.hazard.Hazard
+            The combined hazard object
+        """
+        hazard_list = []
+        for dataset in datasets:
+            target_dir = self._organize_path(dataset, dump_dir) \
+                         if dump_dir == SYSTEM_DIR else dump_dir
+            for dsf in dataset.files:
+                if dsf.file_format == 'hdf5':
+                    hazard_file = self.download_file(target_dir, dsf)
+                    hazard = Hazard()
+                    hazard.read_hdf5(hazard_file)
+                    hazard_list.append(hazard)
+        if not hazard_list:
+            raise ValueError("no hazard files found in datasets")
+
+        hazard_concat = Hazard()
+        hazard_concat = hazard_concat.concat(hazard_list)
```

Comment:
it shouldn't be possible to concat different RCP datasets, I think only countries make sense